### DISCUSSION TITLE
Custom Oauth implementation to support institutional login

### DISCUSSION THREAD

**Discussion by whikloj** [ManifoldScholar/manifold#3750] [Q&A]
Hi,

I have been asked to look at installing Manifold for our institution and the first thing I noticed is that there seems to only be support for local account login or Twitter/Google/Facebook OAuth.

We use Microsoft 365 and have been trying to ensure we have multi-factor authentication in-front of all campus systems.

Does Manifold support a alternative authentication backends and if so, can you point me to the documentation for this?

cheers,
jared

**Comment by whikloj** [comment]
I did see this issue https://github.com/ManifoldScholar/manifold/issues/1075 regarding custom work for CUNY, not sure if this was a one-off and not generalized for others.

**Comment by zdavis** [comment]
This functionality is available for customers of Manifold Digital Services (MDS) — which provides paid, managed Manifold instances.
